source/extensions/tracers/xray/daemon_broker.cc

```diff
@@ -0,0 +1,51 @@
+#include "extensions/tracers/xray/daemon_broker.h"
+
+#include "common/network/utility.h"
+
+#include "source/extensions/tracers/xray/daemon.pb.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace XRay {
+
+namespace {
+// creates a header JSON for X-Ray daemon.
+// For example:
+// { "format": "json", "version": 1}
+std::string createHeader(const std::string& format, uint32_t version) {
```

Comment:
Do we need to validate that format is not empty and version is greater than zero?   Are these checks done elsewhere?

Reply:
hmm - I'm not sure it's necessary. 
It's a helper function instead of hardcoding the (undocumented) X-Ray payload header...
In other words, it's not a function exposed to users of the extension, so I would be validating myself...